### PR TITLE
ENH: Making header_style a property of ExcelFormatter #22758

### LINKED ISSUE #22758: Make header_style a Property of ExcelFormatter
Right now the header style of excel exports is defined as global to the `pandas/io/formats/excel.py` module:

https://github.com/pandas-dev/pandas/blob/1a12c41d201f56439510e683fadfed1218ea9067/pandas/io/formats/excel.py#L37

Because this exists outside of the `ExcelFormatter` class where it is used it is way more difficult to override the default than it needs to be. If this was moved from global scope to instance scope it would allow users to theoretically subclass the `ExcelFormatter` and easily define their own style for the header

This could also eliminate the need for any work on #19386

### PR DESCRIPTION
**Pull request by dannyhyunkim**
- [y] closes #22758
- [y] tests added / passed
- [y] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by WillAyd** [comment]
Nice job! Can you add a whatsnew note for 0.25?

**Comment by WillAyd** [comment]
This is totally my mistake but I meant v0.24 (a whatsnew should already exist). 

You did a great job in adding one but unfortunately that’s not what we want here. Sorry!

**Comment by dannyhyunkim** [comment]
No worries at all. I should have clarified earlier.

**Review by WillAyd — COMMENTED** [review]
I think this looks good but @gfyoung care to take a look?

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -185,6 +185,7 @@ Other Enhancements
 - :class:`Resampler` now is iterable like :class:`GroupBy` (:issue:`15314`).
 - :meth:`Series.resample` and :meth:`DataFrame.resample` have gained the :meth:`Resampler.quantile` (:issue:`15023`).
 - :meth:`Index.to_frame` now supports overriding column name(s) (:issue:`22580`).
+- :class:`ExcelFormatter` now has a `header_style` property. Was previously in global scope (:issue:`22758`).
```
Comment (WillAyd) [inline doc/source/whatsnew/v0.24.0.txt]: Might not actually need this since the ExcelFormatter isn't part of the exposed API

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.24.0.txt]: Should it be? If we want to support use-cases like https://github.com/pandas-dev/pandas/issues/22758 then it should be. OTOH, I don't think we consider things like HTMLFormatter to be public.

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.24.0.txt]: That sounds right to me. (and there's nothing stopping people from using it anyway).

**Comment by gfyoung** [comment]
@WillAyd : Thanks for the ping.  Will take a look later today.

File: doc/source/whatsnew/v0.24.0.txt (hunk 1 above)
Comment (WillAyd) [reply doc/source/whatsnew/v0.24.0.txt]: Opened #22773 to further that discussion; @dannyhyunkim if you can revert the change to whatsnew would be great as this in its current state is more of an internal refactor. Sorry for the back and forth from me on that!

**Review by gfyoung — APPROVED** [review]
LGTM!

cc @jreback @TomAugspurger

**Review by WillAyd — APPROVED** [review]
lgtm as well

**Pull request merged**